src/machine_learning/loss_function/negative_log_likelihood.rs

```diff
@@ -0,0 +1,95 @@
+// Negative Log Likelihood Loss Function
+//
+// The `neg_log_likelihood` function calculates the Negative Log Likelyhood loss,
+// which is a loss function used for classification problems in machine learning.
+//
+// ## Formula
+//
+// For a pair of actual and predicted values, represented as vectors `y_true` and
+// `y_pred`, the Negative Log Likelihood loss is calculated as:
+//
+// - loss = `-y_true * log(y_pred) - (1 - y_true) * log(1 - y_pred)`.
+//
+// It returns the average loss by dividing the `total_loss` by total no. of
+// elements.
+//
+// https://towardsdatascience.com/cross-entropy-negative-log-likelihood-and-all-that-jazz-47a95bd2e81
+
+pub fn neg_log_likelihood(y_true: &[f64], y_pred: &[f64]) -> Option<f64> {
```

Comment:
What do you think this:
```suggestion
pub fn neg_log_likelihood(y_true: &[f64], y_pred: &[f64]) -> Result<f64, NegativeLogLikelihoodLossError> {
```
with
```rust
#[derive(Debug, PartialEq, Eq)]
pub enum NegativeLogLikelihoodLossError {
    InputsHaveDifferentLength,
    EmptyInputs,
}
```

The `Option` should be used in situations when _solution does not exist for some reasonable inputs_. Here the inputs are _obviously_ incorrect.

Reply:
Makes sense for me and just implemented it. But I added one more error which is for invalid values. Is that right or should I change it back?

Like this:
``` python
#[derive(Debug, PartialEq, Eq)]
pub enum NegativeLogLikelihoodLossError {
    InputsHaveDifferentLength,
    EmptyInputs,
    InvalidValues,
}
```